docs/src/main/paradox/java/http/common/http-model.md

```diff
@@ -244,6 +244,28 @@ The actual logic for determining whether to close the connection is quite involv
 request's method, protocol and potential `Connection` header as well as the response's protocol, entity and
 potential `Connection` header. See @github[this test](/akka-http-core/src/test/scala/akka/http/impl/engine/rendering/ResponseRendererSpec.scala#L422) for a full table of what happens when.
 
+<a id="custom-headers-java"></a>
+## Custom Headers
+
+Sometimes you may need to model a custom header type which is not part of HTTP and still be able to use it
+as convenient as is possible with the built-in types.
+
+Because of the number of ways one may interact with headers (i.e. try to convert a `CustomHeader` to a `RawHeader`
+or the other way around etc), a couple of helper classes for custom Header types are provided by Akka HTTP.
+Thanks to extending `ModeledCustomHeader` instead of the plain `CustomHeader` some conversion methods are at your disposal ,
+
+@@snip [CustomHeaderExampleTest.java](../../../../../test/java/docs/http/javadsl/CustomHeaderExampleTest.java) { #modeled-api-key-custom-header }
+
+Which allows the this CustomHeader to be used and created in the following ways:
```

Comment:
Or better, just say "modeled custom header", since you actually talk about the modeled one anyway?